scipy/spatial/transform/_rotation.pyx

```diff
@@ -316,50 +367,69 @@ cdef double[:, :] _compute_euler_from_quat(
             b = quat[ind, i] + quat[ind, k] * sign
             c = quat[ind, j] + quat[ind, 3]
             d = quat[ind, k] * sign - quat[ind, i]
-        
-        # Step 2
-        # Compute second angle...
-        _angles[1] = 2 * atan2(hypot(c, d), hypot(a, b))
-
-        # ... and check if equal to is 0 or pi, causing a singularity
-        if abs(_angles[1]) <= eps:
-            case = 1
-        elif abs(_angles[1] - <double>pi) <= eps:
-            case = 2
-        else:
-            case = 0 # normal case
 
-        # Step 3
-        # compute first and third angles, according to case
-        half_sum = atan2(b, a)
-        half_diff = atan2(d, c)
-        
-        if case == 0:  # no singularities
-            _angles[angle_first] = half_sum - half_diff
-            _angles[angle_third] = half_sum + half_diff
-        
-        else:  # any degenerate case
-            _angles[2] = 0
-            if case == 1:
-                _angles[0] = 2 * half_sum
-            else:
-                _angles[0] = 2 * half_diff * (-1 if extrinsic else 1)
-                
-        # for Tait-Bryan angles
-        if not symmetric:
-            _angles[angle_third] *= sign
-            _angles[1] -= pi / 2
-            
-        for idx in range(3):
-            if _angles[idx] < -pi:
-                _angles[idx] += 2 * pi
-            elif _angles[idx] > pi:
-                _angles[idx] -= 2 * pi
+        _get_angles(angles[ind], extrinsic, symmetric, sign, pi / 2, a, b, c, d)
 
-        if case != 0:
-            warnings.warn("Gimbal lock detected. Setting third angle to zero "
-                          "since it is not possible to uniquely determine "
-                          "all angles.")
+    return angles
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef double[:, :] _compute_davenport_from_quat(
+    np.ndarray[double, ndim=2] quat, np.ndarray[double, ndim=1] n1,
```

Comment:
What if pass `n` as 3x3 by array? Can it simplify things slightly?

Reply:
Up to you, I preferred to pass the axes separately in these low level private functions because I find it easier and more explicit to do just pass them directly after doing the normalization step in the public function, and then the inversion in the case of intrinsic angles.

But if you think it looks better as a single array I can change it

Reply:
I'm totally fine with that approach if you don't see that a single array can make code more simple.

Reply:
Alright, I think I'll leave it like this then!